envoy/upstream/upstream.h

```diff
@@ -1013,7 +1013,7 @@ class ClusterInfo : public Http::FilterChainFactory {
    */
   virtual const envoy::config::cluster::v3::Cluster::CommonLbConfig& lbConfig() const PURE;
 
-  virtual absl::optional<uint64_t>
+  virtual absl::optional<bool>
   processHttpForOutlierDetection(Http::ResponseHeaderMap& reponse) const PURE;
```

Comment:
Rather then adding such new method at ClusterInfo. I think it would be better to expose the `httpProtocolOptions()` here.

Reply:
That is probably a good idea. `ClusterInfo` should be just a config repository without methods with logic.